monkey/common/utils/code_utils.py

```diff
@@ -35,3 +36,18 @@ def del_key(mapping: MutableMapping[T, Any], key: T):
     :param key: A key to delete from `mapping`
     """
     mapping.pop(key, None)
+
+
+def retry(times: int, exceptions: Sequence[Exception]):
+    def inner1(func):
+        @wraps(func)
+        def inner2(*args, **kwargs):
+            for i in range(times):
+                try:
+                    func(*args, **kwargs)
+                except tuple(exceptions):
+                    continue
+
+        return inner2
+
+    return inner1
```

Comment:
Looks nice. Can it be reused in `AutomatedMaster` where we're doing something similar? It might also be beneficial to port over the logging from `AutomatedMaster._try_communicate_with_island()`